### PR TITLE
feat(sdk): add .list_pipeline_versions and .unarchive_experiment methods to Client

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Test code for reviewer, if desired:

```python

from kfp.client import Client

c = Client()

es = c.list_experiments()
experiment_id = 
es.to_dict()['experiments'][0]['id']
print("ARCHIVE", 
c.archive_experiment(experiment_id=experiment_id))
print("GET", 
c.get_experiment(experiment_id=experiment_id))
print("UNARCHIVE", 
c.unarchive_experiment(experiment_id=experiment_id))
print("GET", 
c.get_experiment(experiment_id=experiment_id))

pipeline_id = 
c.list_pipelines().to_dict()['pipelines'][0]['id']
print(c.list_pipeline_versions(pipeline_id=pipeline_id))
```

**Checklist:**
- [x] The title for your pull request (PR) should 
follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/client/client.py [hunk 1]
```diff
@@ -585,16 +585,21 @@ def get_experiment(self,
...
-    def archive_experiment(self, experiment_id: str):
+    def archive_experiment(self, experiment_id: str) -> None:
         """Archives an experiment.
 
         Args:
             experiment_id: id of the experiment.
+        """
+        self._experiment_api.archive_experiment(id=experiment_id)
 
-        Raises:
```
Comment (ji-yaqi) [inline sdk/python/kfp/client/client.py:594]: Do we still have raises part for those?

Comment (connor-mccarthy) [reply sdk/python/kfp/client/client.py:594]: I could not get this to raise an exception, so I removed from the docstring. Test case:

```python
from kfp.client import Client

client = Client()
# does not raise an exception
assert client.archive_experiment(experiment_id="dummy_id") is None
```

Comment (ji-yaqi) [reply sdk/python/kfp/client/client.py:594]: Sounds good, thank you!!

**Pull request merged**